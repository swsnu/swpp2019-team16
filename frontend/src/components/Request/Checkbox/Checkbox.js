import React from 'react';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import PropTypes from 'prop-types';

CheckboxLabels.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  checked: PropTypes.bool.isRequired,
};

export default function CheckboxLabels({ name, value, onClick, checked }) {
  return (
    <FormControlLabel
      data-testid={value}
      control={
        <Checkbox
          name={name}
          onClick={onClick}
          value={value}
          checked={checked}
          color="primary"
        />
      }
      label={value}
    />
  );
}
