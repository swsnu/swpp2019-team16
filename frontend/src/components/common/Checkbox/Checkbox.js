import React from 'react';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import PropTypes from 'prop-types';

CheckboxLabels.propTypes = {
  color: PropTypes.oneOf(['primary', 'secondary']),
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  checked: PropTypes.bool.isRequired,
};

export default function CheckboxLabels({
  name,
  value,
  onClick,
  checked,
  color = 'primary',
}) {
  return (
    <div>
      <FormControlLabel
        data-testid={value}
        control={
          <Checkbox
            name={name}
            onClick={onClick}
            value={value}
            checked={checked}
            color={color}
          />
        }
        label={value}
      />
    </div>
  );
}
