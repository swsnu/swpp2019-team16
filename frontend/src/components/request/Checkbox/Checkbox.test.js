import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Checkbox from './Checkbox';

describe('<Checkbox />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = render(
      <Checkbox children={<div>Checkbox Test</div>} />,
    );

    expect(container).toMatchSnapshot();
  });
});
