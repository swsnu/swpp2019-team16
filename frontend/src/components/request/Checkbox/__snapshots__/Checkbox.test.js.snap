// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> SHOULD match with snapshot 1`] = `
<div>
  <span
    aria-disabled="false"
    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-10 MuiCheckbox-root makeStyles-root-1"
  >
    <span
      class="MuiIconButton-label"
    >
      <input
        class="PrivateSwitchBase-input-13"
        data-indeterminate="false"
        type="checkbox"
        value=""
      />
      <span
        class="makeStyles-icon-2"
      />
    </span>
  </span>
</div>
`;
